### DISCUSSION TITLE
does anybody collect the different middleware performance loss by using skywalking ?

### DISCUSSION THREAD

**Discussion by taojintianxia** [apache/skywalking#7659] [Q&A] [answered]
hi , dear skywalking team

  I tried to search keyword like `performance` or something like that to get the metrics for skywalking performance loss.

  I'm curious how much performance will lost for some middleware like RPC like、mq like、sharding like middleware if we using skywalking and turn the `SW_AGENT_SAMPLE` to 100%.

  I believe the skywalking team must collect the info like that, would you please share the info ? 
or some other people could share the situaion in your company ?

thanks in advance

**Comment by wu-sheng** [comment, marked as answer]
> I believe the skywalking team must collect the info like that, would you please share the info ?

Officially we don't. In theory, all features have their cost for sure.

**Reply by taojintianxia** [reply]
I see, thx
